### DISCUSSION TITLE
Self-hosting: How to overwrite repository specific configs

### DISCUSSION THREAD

**Discussion by amimas** [renovatebot/renovate#10164] [Request Help]
I'm setting up a self-hosted bot. I have the following configurations:

```js
require('dotenv').config()

module.exports = {
  /* Self Hosted configurations */

  platform: 'gitlab',  /* https://docs.renovatebot.com/self-hosted-configuration/#platform */
  endpoint: '****/api/v4/',
  token: process.env.GITLAB_USER_TOKEN,

  autodiscover: true,  /* https://docs.renovatebot.com/self-hosted-configuration/#autodiscover */
  onboarding: false,  /* https://docs.renovatebot.com/self-hosted-configuration/#onboarding */
  requireConfig: false,  /* https://docs.renovatebot.com/self-hosted-configuration/#requireconfig */
  
}
```

I believe the above config will avoid the onboarding process and scan discovered repositories even if it doesn't have a cnofiig file (i.e. `renovate.json`).

How can I configure the bot to ignore configs within those repos (i.e. `renovate.json`)? I'd like all configurations to be managed by the bot. I see there's a [`force` config](https://docs.renovatebot.com/self-hosted-configuration/#force) for overwriting a specific config. There isn't any example. Can I somehow indicate that I want to overwrite all configs (i.e. using `*` pattern) or do I have to list every single configurations for overwriting?

**Comment by rarkins** [comment]
There is a feature request for what you ask here: https://github.com/renovatebot/renovate/issues/6626

**Reply by amimas** [reply]
Thanks @rarkins for the quick reply.

If I read that issue/request correctly, does it mean that having the following in my bot's `config.js` cannot ensure that the bot won't be disabled by setting `enabled: false` from a repo's `renovate.json` file?

```js
force: {
    enabled: true,
  }
```

**Reply by amimas** [reply]
I also need to enforce another setting (i.e. `automerge: false`). Don't want this to be overwritten by a repository.

**Reply by rarkins** [reply]
If you only need to enforce some simple rules (like the boolean ones above) then I think the `force` will work.

**Reply by amimas** [reply]
I see. Thanks again for the quick response.

May I suggest updating the docs around that `force` config. I think others will find it helpful, especially if certain config is not "forcible" at the moment.

https://docs.renovatebot.com/self-hosted-configuration/#force

**Reply by rarkins** [reply]
We don't want to encourage its use, and may deprecate it later - or at least restrict its behavior. And btw the feature request I linked to satisfies your stated requirements to for "all configurations to be managed by the bot", not `force`.

**Reply by rarkins** [reply]
However you're always welcome to submit PRs to improve docs - search the repo and you can find the text you pointed to
